applications/nrf5340_audio/src/bluetooth/ble_audio_services.c

```diff
@@ -272,14 +413,68 @@ int ble_vcs_client_init(void)
 {
 	static struct bt_vcs_cb vcs_client_callback;
 
-	vcs_client_callback.discover = vcs_dis_cb_handler;
+	vcs_client_callback.discover = vcs_discover_cb_handler;
 	vcs_client_callback.state = vcs_state_cb_handler;
 	vcs_client_callback.flags = vcs_flags_cb_handler;
 	return bt_vcs_client_cb_register(&vcs_client_callback);
 }
 #endif /*CONFIG_BT_VCS_CLIENT*/
 #endif /*(CONFIG_AUDIO_DEV == GATEWAY)*/
 
+int ble_mcs_discover(struct bt_conn *conn)
+{
+#if (CONFIG_BT_MCC)
+	return bt_mcc_discover_mcs(conn, true);
+#else
+	LOG_ERR("MCC not enabled");
+	return -ECANCELED;
+#endif /* (CONFIG_BT_MCC) */
+}
+
+int ble_mcs_state_update(struct bt_conn *conn)
+{
+#if (CONFIG_BT_MCC)
+	return bt_mcc_read_media_state(conn);
+#else
+	LOG_ERR("MCC not enabled");
+	return -ECANCELED;
+#endif /* (CONFIG_BT_MCC) */
+}
+
+int ble_mcs_play_pause(struct bt_conn *conn)
+{
+	int ret;
+	struct mpl_cmd cmd;
+
+#if (CONFIG_BT_MCS)
+	media_proxy_ctrl_get_media_state(local_player);
+#endif /* (CONFIG_BT_MCS) */
+
+	if (media_player_state == BT_MCS_MEDIA_STATE_PLAYING) {
+		cmd.opcode = BT_MCS_OPC_PAUSE;
+	} else if (media_player_state == BT_MCS_MEDIA_STATE_PAUSED) {
+		cmd.opcode = BT_MCS_OPC_PLAY;
+	} else {
+		LOG_ERR("Invalid state: %d", media_player_state);
+		return -ECANCELED;
+	}
+
+	cmd.use_param = false;
+
+#if (CONFIG_BT_MCS)
+	ret = media_proxy_ctrl_send_command(local_player, &cmd);
+#elif CONFIG_BT_MCC
+	ret = bt_mcc_send_cmd(conn, &cmd);
+#endif /* (CONFIG_BT_MCS) */
+
+	if (ret) {
+		LOG_ERR("Failed to send play/pause command: %d", ret);
+		return ret;
+	}
+
+	return 0;
+}
+
```

Comment:
ble_vcs_client_init() should be over ble_vcs_server_init() I guess?

Reply:
Agree, also moved it in the .h-file